client/rest/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -50,6 +49,19 @@ public Request(String method, String endpoint) {
         this.endpoint = Objects.requireNonNull(endpoint, "endpoint cannot be null");
     }
 
+    /**
+     * copy the params into {@linkplain Request}.
+     * @param paramSource map holding the properties for request to copy from
+     */
+    public void setParameters(Map<String, String> paramSource){
+        paramSource.keySet().forEach(key -> {
+            if(this.parameters.get(key) != null){
+                throw new IllegalArgumentException(String.format("%s is already set", key));
+            }
+        });
+        this.parameters.putAll(paramSource);
```

Comment:
should this method add parameters to the existing ones or just set them all at once? You could add parameters through `addParameter` and currently we would keep on adding through this new method. But that is not what its name `setParameters` suggests. Shall we rename this to `addParameters` ?

Reply:
If we rename to addParameters we could just make this method call addParameter for each entry in the map, then there is also no duplication around null key check and duplicate check. Does this make sense?